### ISSUE TITLE
Help with adding Bulgarian audio recordings to lessons

### ISSUE THREAD

**Issue by DanWiseProgramming** [Adventech/sabbath-school-lessons#2520]
Hi there,
I would like to add official audio recordings for the Bulgarian Adult Study Guide lessons.
I looked at the audio.yml file for a few of the other languages and they all seem to have the audio split by day.

The recordings produced by Radio 3:16 in Bulgaria are one audio file per week, with a small musical interlude between each day.
Here is a sample: https://web.3-16.bg/lessons/2024_Q1/2024_Q1_Lesson_1.mp3

Would these need to be split into days or is there a way to add these audios straight from the source, with one file for the whole week?

Thanks.
Daniel

**Comment by VitalikL** [comment]
Hi @DanWiseProgramming ,

it is possible to just target the whole week by setting the target for let's say "Sabbath of the week" ex: `target: bg/2024-01/01/01`.

Of course I would say the best user experience would be if we can get split by day. If not, I can update our automatic aggregator to upload audio as they come from the `3-16.bg`. Please note, that it should follow the predictable URL pattern, in this case

 https://web.3-16.bg/lessons/2024_Q1/2024_Q1_Lesson_1.mp3
 https://web.3-16.bg/lessons/2024_Q1/2024_Q1_Lesson_2.mp3
...
 https://web.3-16.bg/lessons/2024_Q1/2024_Q1_Lesson_13.mp3

To summarize: There is a way to do it if we target only Sabbath. If there is no way to get files split by day, we can use that approach. Let me know.

**Comment by DanWiseProgramming** [comment]
Hi @VitalikL ,
Thanks for getting back to me.

If it is not too much trouble, could you please update the automatic aggregator to ingest audio from `web.3-16.bg` for the time being? I will contact the authors of the audios to request that they upload files which are split by day. However, I expect that some time will pass before I get an answer from them.

If they accept and start uploading split audio files, I will open a new issue here notifying you of this so that they can be tagged per-day instead of per-week.
In the event that they refuse, I will look into a solution whereby I can split them myself before providing them to you.

I think that this would be the best course of action for now - what do you think?

**Comment by DanWiseProgramming** [comment]
On an unrelated note, I am a software developer and would like to contribute code where I can - what is the best way to contact you with regards to this matter?

Thanks.

**Comment by VitalikL** [comment]
Thanks @DanWiseProgramming ! 

I setup the auto aggregation targeting weekly audios, hopefully it will be robust. I will monitor the logs in the next few weeks. For now first week audio should be in the app, and once the second week appears it should be added.

Regarding the dev work, we would really welcome any work. There are iOS (swift), Android (kotlin / java), web (js / vue.js / node) that SS app is currently running and most of the code is open source.

**Comment by VitalikL** [comment]
@DanWiseProgramming Looks like the automated script correctly fetched week 2, see this commit: `https://github.com/Adventech/sabbath-school-lessons/commit/3e302c8541695262c7dfa5c5418e9d88ca0b250a`. Therefore closing the ticket. Let me know if at any point there will be possibility of the daily audio

**Comment by DanWiseProgramming** [comment]
Thank you very much @VitalikL ! Will let you know if anything changes with regards to daily audio availability.

God bless you.
